app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java

```diff
@@ -235,15 +246,22 @@ private void monitorSubscription(final ChannelInfo info) {
                 .subscribe(getSubscribeUpdateMonitor(info), onError));
 
         disposables.add(observable
-                // Some updates are very rapid
-                // (for example when calling the updateSubscription(info))
-                // so only update the UI for the latest emission
-                // ("sync" the subscribe button's state)
-                .debounce(100, TimeUnit.MILLISECONDS)
+                .map(List::isEmpty)
+                .distinctUntilChanged()
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe((List<SubscriptionEntity> subscriptionEntities) ->
-                        updateSubscribeButton(!subscriptionEntities.isEmpty()), onError));
+                .subscribe((Boolean isEmpty) -> updateSubscribeButton(!isEmpty), onError));
 
+        disposables.add(observable
+                .map(List::isEmpty)
+                .filter(x -> NotificationHelper.isNewStreamsNotificationsEnabled(requireContext()))
+                .distinctUntilChanged()
+                .skip(1)
```

Comment:
This skips the first subscription status update, i.e. when the channel has just been opened, doesn't it? Please add a comment. What happens if streams notifications are disabled, causing items to stop before arriving at the `skip(1)` call, and then streams notifications are enabled (which could be done without closing the channel fragment since there is a settings button in the menu) and the user subscribes to the channel? That update will be skipped and `showNotifySnackbar()` will never be called. Maybe just put this before `.filter(...)`.

So ultimately I would arrange these lines like this:
```java
.distinctUntilChanged()
.skip(1)
.filter(x -> NotificationHelper.isNewStreamsNotificationsEnabled(requireContext()))
```

Reply:
done